addons/survey/views/survey_survey_views.xml

```diff
@@ -121,7 +121,8 @@
                                             icon="fa-external-link"  target="_blank" class="btn-link pt-0"/>
                                     </div>
                                     <field name="certification_give_badge" attrs="{'invisible': ['|', ('certification', '=', False), ('users_login_required', '=', False)]}" />
-                                    <field name="certification_badge_id" attrs="{'invisible': ['|', ('certification_give_badge', '=', False), ('certification_badge_id', '!=', False)]}"
+                                    <field name="certification_badge_id"
+                                           attrs="{'invisible': ['|', ('certification_give_badge', '=', False), ('certification_badge_id', '!=', False)], 'required': [('certification_give_badge', '=', True), ('certification_badge_id', '=', False)]}"
```

Comment:
required : not sure ('certification_badge_id', '=', False) is necessary.
if give_badge = True, certification_badge_id should be set, and that's it.
unless I am missing something ?